storm-core/src/jvm/backtype/storm/scheduler/resource/strategies/priority/ISchedulingPriorityStrategy.java

```diff
@@ -16,22 +16,26 @@
  * limitations under the License.
  */
 
-package backtype.storm.scheduler.resource.strategies;
-
-import java.util.Collection;
-import java.util.Map;
+package backtype.storm.scheduler.resource.strategies.priority;
 
+import backtype.storm.scheduler.Cluster;
 import backtype.storm.scheduler.Topologies;
-import backtype.storm.scheduler.ExecutorDetails;
 import backtype.storm.scheduler.TopologyDetails;
-import backtype.storm.scheduler.WorkerSlot;
-import backtype.storm.scheduler.resource.RAS_Node;
+import backtype.storm.scheduler.resource.RAS_Nodes;
+import backtype.storm.scheduler.resource.User;
 
-/**
- * An interface to for implementing different scheduling strategies for the resource aware scheduling
- * In the future stategies will be pluggable
- */
-public interface IStrategy {
+import java.util.Map;
+
+public interface ISchedulingPriorityStrategy {
+
+    /**
+     * initializes
+     */
+    public void prepare(Topologies topologies, Cluster cluster, Map<String, User> userMap, RAS_Nodes nodes);
```

Comment:
well we cannot get all the topologydetails from cluster.  And we would have to recreate all the node objects from cluster which I think might be unnecessary since we already do it in the scheduler so why do this computation again uneccessarily